kimera_vio_ros/src/interfaces/base_interface.cpp

```diff
@@ -0,0 +1,74 @@
+#include <chrono>
+
+#include "kimera_vio_ros/interfaces/base_interface.hpp"
+
+using namespace std::chrono_literals;
+
+namespace kimera_vio_ros
+{
+namespace interfaces
+{
+
+BaseInterface::BaseInterface(
+  rclcpp::Node::SharedPtr & node)
+: VIO::DataProviderInterface(),
```

Comment:
@ToniRV , Instead of inheriting from the `VIO::DataProviderInterface` class, It might be cleaner to replicate what was changed in the ROS1 wrapper and create a `data_provider_` filed:

* https://github.com/MIT-SPARK/Kimera-VIO-ROS/blob/ec5f3d4325612b4526f81e3314bda1c38836ae59/src/KimeraVioRos.cpp#L36

However, sterio and imu child interfaces here wouldn't then be able to access the private vio callback functions:

* https://github.com/MIT-SPARK/Kimera-VIO/blob/d677a3dedad2da605c84a4b7f51154cd7e38a98d/include/kimera-vio/dataprovider/DataProviderInterface.h#L77-L83

I tried creating a protected child class `VIO::DataProviderInterface`, and listing `BaseInterface` as a protected friend, but that workaround didn't seem to compile. It might of just have been my error in playing with the C++ classes.